gammapy/analysis/core.py

```diff
@@ -204,55 +165,88 @@ def get_flux_points(self, source="source"):
         if not self._validate_fp_settings():
             return False
 
+        fp_settings = self.config.flux_points
         # TODO: add "source" to config
         log.info("Calculating flux points.")
-        axis_params = self.settings["flux-points"]["fp_binning"]
-        e_edges = MapAxis.from_bounds(**axis_params).edges
+        e_edges = self._make_energy_axis(fp_settings.energy).edges
         flux_point_estimator = FluxPointsEstimator(
             e_edges=e_edges, datasets=self.datasets, source=source
         )
         fp = flux_point_estimator.run()
         fp.table["is_ul"] = fp.table["ts"] < 4
-        model = self.model[source].copy()
-        self.flux_points = FluxPointsDataset(data=fp, model=model)
+        self.flux_points = FluxPointsDataset(data=fp, model=self.model[source])
         cols = ["e_ref", "ref_flux", "dnde", "dnde_ul", "dnde_err", "is_ul"]
         log.info("\n{}".format(self.flux_points.data.table[cols]))
 
-    @staticmethod
-    def _create_geometry(params):
+    def update_config(self, config=None, filename=""):
```

Comment:
Can we move `analysis.update_config` to `analysis.config.update`?

If you look at the current `analysis.update_config` you'll see that it only accesses `self.config`, nothing else from `self`, i.e. it only interacts with the config.

Or is there a technical problem with this suggestion, e.g. because it's not clear how to do an in-place update of the values? Then maybe the method could still be on the config class, and the analysis class only has a wrapper like this?
```
class Analysis:
    def update_config(self, config):
        self.config = self.config.update(config)
```